maldoca/ole/oss_utils.cc

```diff
@@ -434,10 +426,83 @@ bool BufferToUtf8::ConvertEncodingBufferToUTF8String(absl::string_view input,
   }
   CHECK(internal_converter_ == InternalConverter::kNone);
   size_t in_bytes_left = input.size();
+  DLOG(INFO) << "input.size: " << input.size();
   if (in_bytes_left == 0) {
     return true;
   }
   const char* input_ptr = input.data();
+
+#if defined(_WIN32)
+  int mb_size = 0;
+  int wc_size = 0;
+  wchar_t* w_str;
+  bool is_already_utf16 = code_page_ == 1200;
+
+  // only works for ascii??
+  /*if (is_already_utf16) {
+    out_str->resize(input.size());
+    const wchar_t* input_wptr = (wchar_t*) input.data();
+    mb_size = wcstombs(&(*out_str)[0], input_wptr, input.size());
+  } else {*/
+
+  // No need to convert to UTF-16, if the string is already UTF-16 encoded.
+  if (!is_already_utf16) {
+    // Calculate output size in UTF-16.
+    wc_size =
+        MultiByteToWideChar(code_page_, 0, input_ptr, input.size(), NULL, 0);
+    DLOG(INFO) << "Output size in UTF-16: " << wc_size;
+    if (wc_size <= 0) {
+      LOG(ERROR) << "Error while calculating output size in UTF-16: "
+                 << GetLastError();
+    }
+
+    // Convert input to UTF-16.
+    w_str = new wchar_t[wc_size];
+    wc_size = MultiByteToWideChar(code_page_, 0, input_ptr, input.size(), w_str,
+                                  wc_size);
+    if (wc_size <= 0) {
+      LOG(ERROR) << "Error while converting to UTF-16: " << GetLastError();
+    }
+  } else {
+    w_str = (wchar_t*)input.data();
+    wc_size = input.size();
```

Comment:
In this case, shouldn't this be input.size() / sizeof(wchar_t) since it's supposed to the "Returns the number of characters written to the buffer"  according to https://blogs.msmvps.com/gdicanio/2017/02/03/the-new-c11-u16string-doesnt-play-well-with-win32-apis/  while input.size is # in bytes.

Reply:
oops not /sizeof(wchar_t) but just  >> 1 since this is the # of unicode which is 2 bytes each even though on windows the two are the same.

Reply:
done